### PR TITLE
RUBY-1657 added error message to timeouts

### PR DESCRIPTION
**Pull request by Neilshweky**
I couldn't figure out how to test for this reliably. I tried to set the timeout duration from the test but ultimately it didn't work...

### CODE REVIEW

File: lib/mongo/socket/ssl.rb [hunk 1]
```diff
@@ -137,7 +137,7 @@ def initialize(host, port, host_name, timeout, family, options = {})
       #
       # @since 2.0.0
       def connect!
-        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError) do
+        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError, 'the socket took too long to connect') do
```
Comment (p-mongo) [inline lib/mongo/socket/ssl.rb:140]: Can you please add the timeout into the message, e.g. `The socket took over #{options[:connect_timeout]} seconds to connect` (and capitalize the first letter).

**Comment by p-mongo** [comment]
I don't think we need test coverage for this change.

**Comment by Neilshweky** [comment]
Great! Is there anything else you need me to add then?

**Comment by p-mongo** [comment]
@Neilshweky Can you please add the timeout values to the messages as I mentioned in the comment?

**Comment by Neilshweky** [comment]
@p-mongo Sorry I didn't see your comment, I just changed that and pushed it

File: lib/mongo/socket/ssl.rb [hunk 2]
```diff
@@ -137,7 +137,7 @@ def initialize(host, port, host_name, timeout, family, options = {})
       #
       # @since 2.0.0
       def connect!
-        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError) do
+        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError, 'The socket took over #{options[:connect_timeout]} seconds to connect') do
```
Comment (p-mongo) [inline lib/mongo/socket/ssl.rb:140]: The string needs double quotes to perform interpolation.

You can test the message that is generated by adding e.g. a `sleep 10` call in the block here, and instantiating a client with a lower connect timeout: `Mongo::Client.new(['localhost'], connect_timeout: 1)`. This should raise the timeout error with the message you are adding here.

Comment (Neilshweky) [reply lib/mongo/socket/ssl.rb]: fixed and tested.

**Pull request merged**